cdap-integration-test/src/main/java/co/cask/cdap/test/IntegrationTestBase.java

```diff
@@ -150,15 +158,13 @@ public Boolean call() throws Exception {
 
 
   private void checkServicesWithRetry(Callable<Boolean> callable,
-                                      String exceptionMessage) throws TimeoutException {
-    int numSecs = 0;
+                                      String exceptionMessage) throws TimeoutException, InterruptedException {
```

Comment:
If Sleep is interrupted, that's nothing to do with the callable logic or CDAP; it means something else interrupted the thread, which I think we should not retry in that case.